### PR TITLE
Contract events subscription

### PR DESCRIPTION
**Pull request by Lbqds**

### CODE REVIEW

File: src/events.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      if (!this.cancelled) {
+        this.task = setTimeout(() => this.eventEmitter.emit('tick'), this.pollingInterval)
+      }
+    } catch (err) {
+      this.errorCallback(err, this)
+    }
+  }
+}
+
+export function subscribe(options: SubscribeOptions): Subscription {
```
Comment (Lbqds) [inline src/events.ts:103]: Maybe we can move this to `contract.ts`.

File: src/events.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  currentEventCount(): number {
+    return this.fromCount
+  }
+
+  private async fetchEvents() {
+    try {
+      const response = await this.client.events.getEventsContractContractaddress(this.contractAddress, {
+        start: this.fromCount
+      })
```
Comment (polarker) [inline src/events.ts:80]: We could use `convertHttpResponse` here

File: src/events.ts (hunk 1 above)
Comment (polarker) [reply src/events.ts:104]: Noted. I will restructure the whole project today or tomorrow

**Review by polarker — APPROVED** [review]
LGTM. I will merge after the signer PR

**Pull request merged**